scan.py

```diff
@@ -205,13 +206,19 @@ def add_grocery_item(trello_api, item):
     except urllib2.HTTPError, e:
         if 'UPC/EAN code invalid' in e.msg:
             print "Barcode {0} not recognized as a UPC; creating learning opportunity".format(repr(barcode))
-            opp = create_barcode_opp(trello_db, barcode)
+            try:
+                opp = create_barcode_opp(trello_db, barcode, desc)
+            except:
```

Comment:
Under what kind of circumstances would this fail and need to be retried like this?